commit 7c565501e91f3759274dd7cd43ec9de7c4b9a162
Author: William Lallemand <wlallemand@haproxy.org>
Date:   Wed Dec 29 18:16:27 2021 +0100

    REGTESTS: ssl: fix ssl_default_server.vtc
    
    Patch 2c776f1 ("BUG/MEDIUM: ssl: initialize correctly ssl w/
    default-server") added tests that are not relevant anymore and broke the
    reg-test. revert them.
    
    (cherry picked from commit 0387632ac0db520402550cf19a96d41f8c1661de)
    Signed-off-by: Willy Tarreau <w@1wt.eu>
    (cherry picked from commit 5229b2badbc370ef11ad5c6a7c9529bd24b2de66)
    Signed-off-by: Willy Tarreau <w@1wt.eu>

--- a/reg-tests/ssl/ssl_default_server.vtc
+++ b/reg-tests/ssl/ssl_default_server.vtc
@@ -15,7 +15,7 @@ feature cmd "$HAPROXY_PROGRAM -cc 'versi
 feature cmd "$HAPROXY_PROGRAM -cc 'feature(OPENSSL)'"
 feature ignore_unknown_macro
 
-server s1 -repeat 10 {
+server s1 -repeat 7 {
   rxreq
   txresp
 } -start
@@ -56,10 +56,7 @@ haproxy h1 -conf {
 
     backend third_be
         default-server ssl crt client1.pem ca-file ca-auth.crt verify none
-        server s1 "${tmpdir}/ssl.sock"
-        server s2 "${tmpdir}/ssl.sock" crt client2_expired.pem
-        server s3 "${tmpdir}/ssl.sock"
-        server s4 "${tmpdir}/ssl.sock"
+        server s1 "${tmpdir}/ssl.sock" crt client2_expired.pem
 
     backend fourth_be
         default-server ssl crt client1.pem verify none
@@ -109,25 +106,9 @@ client c1 -connect ${h1_clearlst_sock} {
     txreq
     rxresp
     expect resp.status == 200
-    expect resp.http.x-ssl == "Ok"
-} -run
-
-client c1 -connect ${h1_clearlst_sock} {
-    txreq -url "/third"
-    txreq
-    rxresp
-    expect resp.status == 200
     expect resp.http.x-ssl == "Expired"
 } -run
 
-client c1 -connect ${h1_clearlst_sock} -repeat 2 {
-    txreq -url "/third"
-    txreq
-    rxresp
-    expect resp.status == 200
-    expect resp.http.x-ssl == "Ok"
-} -run
-
 client c1 -connect ${h1_clearlst_sock} {
     txreq -url "/fourth"
     txreq
